lib/java/src/org/apache/thrift/transport/TFramedTransport.java

```diff
@@ -65,17 +65,26 @@ public TTransport getTransport(TTransport base) {
     }
   }
 
+  /**
+   * Something to fill in the first four bytes of the buffer
+   * to make room for the frame size.  This allows the
+   * implementation to write once instead of twice.
+   */
+  private static final byte[] sizeFiller_ = new byte[] { 0x00, 0x00, 0x00, 0x00 };
+
   /**
    * Constructor wraps around another transport
    */
   public TFramedTransport(TTransport transport, int maxLength) {
     transport_ = transport;
     maxLength_ = maxLength;
+    writeBuffer_.write(sizeFiller_, 0, 4);
```

Comment:
Q: why were you able to do the initial write in the constructor here, but not in `TFastFramedTransport`?

Reply:
The buffer classes are different and write buffer here does not throw TTransportException.